### PR TITLE
Closes #2306 Delete renv

### LINKED ISSUE #2306: General Issue:  Remove `renv` from admiral
### Background Information

It has been decided to remove renv from how to develop the admiral R package.  

renv did help establish a common framework for us all to work in, but added a lot of complications to our processes as well as was a bit of a hurdle for new developers coming onboard - either new to pgk dev or coming form other R packages

### Definition of Done

* Remove the `renv` folder
* Remove the `renv.lock` file
* Please make a blurb in the news as it is a big milestone for us.
* Investigate any issues that come up from renv removal and report back to the team

@pharmaverse/admiral @pharmaverse/admiral_comm A simple file/folder deletion with a quick blurb in the news.  There might be some unknown downstream issues we might have to investigate...but that will be probably not be apart of this Issue/PR

### PR DESCRIPTION
**Pull request by dgrassellyb**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF version of it to the same folder.
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by dgrassellyb** [comment]
@bms63 / @ddsjoberg / @esimms999-gsk : just solved some issues with the worfklows (you can merge this one inside 2304_add_unit_tests_get_summary_records branch)

**Comment by bms63** [comment]
I think something is amiss as we lost all our CI checks?

Do you mind adding a news blurb as it is a big milestone for us!! :)

File: README.md
Comment (bms63) [inline README.md:1]: Can we fully remove the `renv` folder?  I still see it in this branch.
image

Also do we need to adjust `.Rprofile` 

image

File: NEWS.md
Comment (bms63) [inline NEWS.md:1]: Can you move this up to the dev version section of the news please

Comment (dgrassellyb) [reply NEWS.md:1]: @bms63 : renv is well deleted and .Rprofile updated as well ? https://github.com/pharmaverse/admiral/pull/2307/files

File: README.md
Comment (dgrassellyb) [reply README.md:1]: @bms63 : renv is well deleted and .Rprofile updated as well ? https://github.com/pharmaverse/admiral/pull/2307/files

File: NEWS.md
Comment (dgrassellyb) [reply NEWS.md:1]: yes I'll do !

File: .Rprofile [hunk 1]
```diff
@@ -1,47 +1,45 @@
...
+      Sys.setenv("RENV_PROFILE" = .renv_profile)
+    } else {
+      message("This repository do not contains the renv profile for your R version.")
+    }
   } else {
-    message("This repository do not contains the renv profile for your R version.")
+    message(
+      "Using renv profile from `renv/profile` file.\n",
+      "The `", readLines("./renv/profile"), "` profile will be used."
+    )
```
Comment (bms63) [inline .Rprofile:39]: Isn't this going to search for something that doesn't exist anymore??

**Comment by bms63** [comment]
I pulled your branch down adn still see this `renv` folder.  I don't think it is fully removed.

image

**Comment by bms63** [comment]
Can we remove this messaging as well - it is confusing - happens when I restart the session 
image

**Comment by ddsjoberg** [comment]
@bms63  It looks like the renv folder is removed. Here's what the repo looks like on that branch: 
https://github.com/pharmaverse/admiral/tree/d53ea3a0fbbd7f64505cd0e2cb564c8666041c61

git won't delete a folder that exists on your machine (even if every file i the folder has been removed from the repo)

**Comment by bms63** [comment]
> @bms63 It looks like the renv folder is removed. Here's what the repo looks like on that branch: https://github.com/pharmaverse/admiral/tree/d53ea3a0fbbd7f64505cd0e2cb564c8666041c61
> 
> git won't delete a folder that exists on your machine (even if every file i the folder has been removed from the repo)

Interesting - I am wondering if maybe it is making the folder when due to the `.Rprofile` setup? 
image

File: .Rprofile (hunk 1 above)
Comment (dgrassellyb) [reply .Rprofile:39]: yes .. actually it's well configured for github actions but I didn't check for local config, but the question is what would you like to have exactly for local R sessions ? (I am not sure to fully understand how you want to work locally - we will have both cases people using renv and some others using devtools .. ? (I am not even sure we need to customize this file anymore to be honest) .. what do you think @cicdguy ?

**Comment by dgrassellyb** [comment]
(I will tag this MR as draft one @bms63 - still need some discussions + figured out we have a problem with roxygen version)

**Review by bms63 — COMMENTED** [review]
@dgrassellyb that is a lot of files changes - all expected?  can we merge this into @esimms999-gsk PR as that is where this PR is pointing??

**Comment by bms63** [comment]
@dgrassellyb something happened to the checks again after the common was updated - was that intended?

**Comment by dgrassellyb** [comment]
> @dgrassellyb something happened to the checks again after the common was updated - was that intended?

Yes it's because the target branch is not main branch (but I manually triggered actions to be sure : https://github.com/pharmaverse/admiral/actions/runs/7651814366 but we just do not see them on checks when target branch is not main)

**Pull request merged**